de.dlr.sc.virsat.server/src/de/dlr/sc/virsat/server/controller/RepoManagementController.java

```diff
@@ -0,0 +1,55 @@
+/*******************************************************************************
+ * Copyright (c) 2008-2019 German Aerospace Center (DLR), Simulation and Software Technology, Germany.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package de.dlr.sc.virsat.server.controller;
+
+import java.io.File;
+import java.net.URISyntaxException;
+
+import org.eclipse.core.runtime.Status;
+
+import de.dlr.sc.virsat.server.Activator;
+import de.dlr.sc.virsat.server.configuration.RepositoryConfiguration;
+import de.dlr.sc.virsat.server.configuration.ServerConfiguration;
+import de.dlr.sc.virsat.server.repository.RepoRegistry;
+import de.dlr.sc.virsat.server.repository.ServerRepository;
+
+public class RepoManagementController {
+	
+	public RepoManagementController() {
+	}
+	
+	public ServerRepository getRepository(String repoName) {
+		return RepoRegistry.getInstance().getRepository(repoName);
+	}
+	
+	/**
+	 * Adds and registers a new project
+	 * @param repoConfiguration project configuration
+	 */
+	public void addNewRepository(RepositoryConfiguration repoConfiguration) {
+		try {
+			ServerRepository repository = new ServerRepository(new File(ServerConfiguration.getProjectRepositoriesDir()), repoConfiguration);
+			RepoRegistry.getInstance().addRepository(repoConfiguration.getProjectName(), repository);
+		} catch (URISyntaxException e) {
+			String errorMessage = "Error adding repository configuration for project " + repoConfiguration.getProjectName();

```

Comment:
Is it ok that we catch the exception here but then return Response.Status.OK in the calling method?

Reply:
No, this is definitely wrong :) thanks!